src/shogun/evaluation/MachineEvaluation.cpp

```diff
@@ -121,5 +161,22 @@ CMachine* CMachineEvaluation::get_machine() const
 
 EEvaluationDirection CMachineEvaluation::get_evaluation_direction()
 {
-	return m_evaluation_criterion->get_evaluation_direction();
+	REQUIRE(m_evaluation_criteria->get_array_size()==1,
+		"Multiple Metrics provided. Please use get_evaluation_directions()");
+	CEvaluation* temp_eval = (CEvaluation*) m_evaluation_criteria->get_element_safe(0);
```

Comment:
there might be a memory leak here as 
```m_evaluation_criteria->get_element_safe(0);``` increases the reference counter by one (I think)
If it does (check code), then you have to do ```SG_UNREF(temp_eval);``` in order to avoid leaks